src/include/planner/plannodes/create_table_plan_node.h

```diff
@@ -595,15 +597,15 @@ class CreateTablePlanNode : public AbstractPlanNode {
      * @param col multi-column constraint definition
      * @return builder object
      */
-    Builder &ProcessCheckConstraint(const std::shared_ptr<parser::ColumnDefinition> &col) {
+    Builder &ProcessCheckConstraint(const std::unique_ptr<parser::ColumnDefinition> &col) {
       auto check_cols = std::vector<std::string>();
 
       // TODO(Gus,Wen) more expression types need to be supported
       if (col->GetCheckExpression()->GetReturnValueType() == type::TypeId::BOOLEAN) {
         check_cols.push_back(col->GetColumnName());
 
-        parser::ConstantValueExpression *const_expr_elem =
-            dynamic_cast<parser::ConstantValueExpression *>(col->GetCheckExpression()->GetChild(1).get());
+        auto const_expr_elem =
```

Comment:
why change this to auto? Using auto makes code harder to read.

Reply:
We have the type on the RHS with the .CastManagedPointerTo<Type>, but I have no objection to writing it out. Changed.